src/DoctrineBatchUtils/BatchProcessing/SelectBatchIteratorAggregate.php

```diff
@@ -20,32 +19,49 @@
  * 'Read' focused batching iterator that will issue a clear to the entity manager
  * every batch size to 'detach' the managed objects and therefore make them entities only for the
  * purpose of reading without manually attaching them back to the EntityManager.
+ *
+ * @template A
```

Comment:
Let's call these TKey, TValue rather than A and B

Reply:
I think `A` should be constrained to `int|string`

Reply:
If i constraint TKey to int|string i got this error

``` 
ERROR: MixedArgumentTypeCoercion - src/DoctrineBatchUtils/BatchProcessing/SelectBatchIteratorAggregate.php:39:25 - Argument 1 of DoctrineBatchUtils\BatchProcessing\SelectBatchIteratorAggregate::__construct expects iterable<int|string, mixed>, parent type iterable<mixed, mixed> provided (see https://psalm.dev/194)
        return new self($query->toIterable(), $query->getEntityManager(), $batchSize)
```